api/v1alpha1/oadp_types.go

```diff
@@ -231,97 +223,8 @@ type SnapshotLocation struct {
-	// +optional
-	SnapshotRetainPolicy *RetainPolicy `json:"snapshotRetainPolicy,omitempty"`
-	// schedule is a cronspec (https://en.wikipedia.org/wiki/Cron#Overview) that
-	// can be used to schedule datamover(volsync) synchronization to occur at regular, time-based
-	// intervals. For example, in order to enforce datamover SnapshotRetainPolicy at a regular interval you need to
-	// specify this Schedule trigger as a cron expression, by default the trigger is a manual trigger. For more details
-	// on Volsync triggers, refer: https://volsync.readthedocs.io/en/stable/usage/triggers.html
-	//+kubebuilder:validation:Pattern=`^(\d+|\*)(/\d+)?(\s+(\d+|\*)(/\d+)?){4}$`
-	//+optional
-	Schedule string `json:"schedule,omitempty"`
-}
-
-// RetainPolicy defines the fields for retention of datamover snapshots
-type RetainPolicy struct {
-	// Hourly defines the number of snapshots to be kept hourly
-	//+optional
-	Hourly string `json:"hourly,omitempty"`
-	// Daily defines the number of snapshots to be kept daily
-	//+optional
-	Daily string `json:"daily,omitempty"`
-	// Weekly defines the number of snapshots to be kept weekly
-	//+optional
-	Weekly string `json:"weekly,omitempty"`
-	// Monthly defines the number of snapshots to be kept monthly
-	//+optional
-	Monthly string `json:"monthly,omitempty"`
-	// Yearly defines the number of snapshots to be kept yearly
-	//+optional
-	Yearly string `json:"yearly,omitempty"`
-	// Within defines the number of snapshots to be kept Within the given time period
-	//+optional
-	Within string `json:"within,omitempty"`
-}
-
-type DataMoverVolumeOptions struct {
-	SourceVolumeOptions      *VolumeOptions `json:"sourceVolumeOptions,omitempty"`
-	DestinationVolumeOptions *VolumeOptions `json:"destinationVolumeOptions,omitempty"`
-}
-
-// VolumeOptions defines configurations for VolSync options
-type VolumeOptions struct {
-	// storageClassName can be used to override the StorageClass of the source
-	// or destination PVC
-	//+optional
-	StorageClassName string `json:"storageClassName,omitempty"`
-	// accessMode can be used to override the accessMode of the source or
-	// destination PVC
-	//+optional
-	AccessMode corev1.PersistentVolumeAccessMode `json:"accessMode,omitempty"`
-	// cacheStorageClassName is the storageClass that should be used when provisioning
-	// the data mover cache volume
-	//+optional
-	CacheStorageClassName string `json:"cacheStorageClassName,omitempty"`
-	// cacheCapacity determines the size of the restic metadata cache volume
-	//+optional
-	CacheCapacity string `json:"cacheCapacity,omitempty"`
-	// cacheAccessMode is the access mode to be used to provision the cache volume
-	//+optional
-	CacheAccessMode string `json:"cacheAccessMode,omitempty"`
-}
-
 // Features defines the configuration for the DPA to enable the tech preview features
-type Features struct {
-	// Contains data mover specific configurations
-	// +optional
-	DataMover *DataMover `json:"dataMover,omitempty"`
-}
+type Features struct{}
```

Comment:
I think we discussed this: left it as an empty struct, right?

Reply:
I think empty struct would still cause breakage.

We would want to in this types.go, whereever this is referenced to omit empty  in the jsontag for at leats one minor version ie. oadp 1.3.z

then we warn/notify user to remove so they know about velero upstream data mover.

Reply:
The reason for the empty struct is that we may (in the future) have additional Features to add.

Reply:
resolving this conversation as we agree